File: repos_estudiantes_ALVAREZ_primera_entrega/APAZA_PACHECO_BRYAN_ORLANDO/Polimorfismo/Ejer7/Parada.py
# a) 
class Parada:
    def __init__(self,nombreP=""):
        self.admins=[""]*10
        self.autos=[["",""] for _ in range(10)]
        self.nombreP=nombreP
        self.nroAdmins=0
        self.nroAutos=0

    # b) 
    def mostrar(self):
        print("Parada:",self.nombreP)
        print("Admins:")
        for i in range(self.nroAdmins):
            print("-",self.admins[i])
        print("Autos:")
        for i in range(self.nroAutos):
            print("Modelo:",self.autos[i][0],"Conductor:",self.autos[i][1])

    # c) 
    def adicionar(self,x):
        if self.nroAdmins<10:
            self.admins[self.nroAdmins]=x
            self.nroAdmins+=1
        else:
            print("No se pueden agregar más admins")

    # d) 
    def adicionar_auto(self,modelo,conductor):
        if self.nroAutos<10:
            self.autos[self.nroAutos][0]=modelo
            self.autos[self.nroAutos][1]=conductor
            self.nroAutos+=1
        else:
            print("No se pueden agregar más autos")

p1=Parada("Terminal")
p1.adicionar("Bryan")
p1.adicionar("Kevin")
p1.adicionar_auto("Toyota","Luis")
p1.adicionar_auto("Nissan","Juan")
p1.mostrar()

File: repos_estudiantes_ALVAREZ_primera_entrega/CHAMBI_BARRETO_CARLOS_ERNESTO/POLIMORFISMO/EJERCICIO7/Parada.java
package POLIMORFISMO.EJERCICIO7;

public class Parada {
    private String[] admins;
    private String[][] autos;
    private String nombreP;
    private int nroAdmins, nroAutos;

    public Parada() {
        this.admins =  new String[10];
        this.autos = new String[10][3];
        this.nombreP = "sin especificaf";
        this.nroAdmins = 0;
        this.nroAutos = 0;
    }
    public Parada(String nombreP) {
        this();
        this.nombreP = nombreP;
    }
    public void mostrar(){
        System.out.println("Parara: "+ nombreP);
        System.out.println("Administradores");
        for (int i = 0 ; i< nroAdmins ; i++){
            System.out.println(admins[i]);
        }
        System.out.println("Autos");
        for(int i = 0 ; i < nroAutos ; i++){
            System.out.println("Modelo: " + autos[i][0] +
                    ", Conductor: " + autos[i][1] +
                    ", Placa: " + autos[i][2]);
        }
        System.out.println("--------------------------");

    }
    public void adicionar(String admin) {
        if (nroAdmins < admins.length) {
            admins[nroAdmins] = admin;
            nroAdmins++;
        } else {
            System.out.println("No se pueden agregar más administradores.");
        }
    }
    public void adicionar(String modelo, String conductor, String placa) {
        if (nroAutos < autos.length) {
            autos[nroAutos][0] = modelo;
            autos[nroAutos][1] = conductor;
            autos[nroAutos][2] = placa;
            nroAutos++;
        } else {
            System.out.println("No se pueden agregar más autos.");
        }
    }
}
